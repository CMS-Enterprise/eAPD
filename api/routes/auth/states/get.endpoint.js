const { actualVerifyEAPDToken } = require('../../../auth/jwtUtils');

const {
  getDB,
  setupDB,
  teardownDB,
  login,
  unauthenticatedTest,
  unauthorizedTest
} = require('../../../endpoint-tests/utils');

describe('auth roles endpoint get endpoint', () => {
  describe('GET /auth/state/:stateId', () => {
    const db = getDB();
    beforeAll(() => setupDB(db));
    afterAll(() => teardownDB(db));
    const url = '/auth/state';

    unauthenticatedTest('get', `${url}/ak`);
    unauthorizedTest('get', `${url}/ak`);

    it('when authenticated', async () => {
      const api = login();
      const response = await api.get(`${url}/md`);

      expect(response.status).toEqual(200);
      const claims = actualVerifyEAPDToken(response.data.jwt);
      expect(claims.state).toMatchSnapshot();
    });
  });
});
