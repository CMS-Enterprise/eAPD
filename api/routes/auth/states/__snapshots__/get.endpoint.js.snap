// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth roles endpoint get endpoint GET /auth/state/:stateId when authenticated 1`] = `
Object {
  "id": "md",
  "medicaid_office": null,
  "name": "Maryland",
}
`;
